mclocations/minecraft.py

```diff
@@ -0,0 +1,89 @@
+import re
+
+from mcrcon import MCRcon
+
+from mclocations.models import Player, Position
+
+
+class MinecraftServer:
+    def __init__(self, host, password, port=25575):
+        self.mcr = MCRcon(host, password, port)
+        self.mcr.connect()
+
+    def disconnect(self):
+        self.mcr.disconnect()
+
+    def get_players(self):
+        players = []
+        player_names = MinecraftServer._parse_list(self.mcr.command('/list'))
+
+        for player_name in player_names:
+            position = MinecraftServer._parse_position(self.mcr.command(f'/data get entity {player_name} Pos'))
+            dimension = MinecraftServer._parse_dimension(self.mcr.command(f'/data get entity {player_name} Dimension'))
+            players.append(Player(player_name, position, dimension))
+
+        return players
+
+    @classmethod
+    def _parse_list(cls, message):
+        """
+        >>> MinecraftServer._parse_list("There are 0 of a max of 20 players online:")
+        []
+        >>> MinecraftServer._parse_list("There are 1 of a max of 20 players online: _DavidCoe_")
+        ['_DavidCoe_']
+        >>> MinecraftServer._parse_list("There are 2 of a max of 20 players online: _DavidCoe_, _NotDavidCoe_")
+        ['_DavidCoe_', '_NotDavidCoe_']
+
+        :param message:
+        :return:
+        """
+        match = re.match(
+            r'There are (?P<current_count>\d+) of a max of (?P<max_count>\d+) players online:(?: (?P<players>.+))',

```

Comment:
Interesting tidbit, this message was changed in 1.16 so this will not work for <=1.15.2... you want to know what changed?

**1.16**
> There are 0 of a max of 20 players online:

**1.15.2**
> There are 0 of a max 20 players online:

Super awesome and definitely required change there huh?

Reply:
Yuck!